core/lib/zksync_core/src/consensus/testonly.rs

```diff
@@ -235,72 +278,47 @@ impl StateKeeperHandle {
         }
     }
 
-    // Wait for all pushed miniblocks to be produced.
-    pub async fn sync(&self, ctx: &ctx::Ctx, pool: &ConnectionPool) -> anyhow::Result<()> {
-        const POLL_INTERVAL: time::Duration = time::Duration::milliseconds(100);
-        loop {
-            let mut storage = pool.access_storage().await.context("access_storage()")?;
-            let head = storage
-                .blocks_dal()
-                .get_sealed_miniblock_number()
-                .await
-                .context("get_sealed_miniblock_number()")?;
-            if head.0 >= self.next_block.0 - 1 {
-                return Ok(());
-            }
-            ctx.sleep(POLL_INTERVAL).await?;
-        }
+    /// Last block that has been pushed to the `StateKeeper` via `ActionQueue`.
+    /// It might NOT be present in storage yet.
+    pub fn last_block(&self) -> validator::BlockNumber {
+        validator::BlockNumber(self.last_block.0 as u64)
     }
 
-    // Wait for all pushed miniblocks to have consensus certificate.
-    pub async fn sync_consensus(
-        &self,
-        ctx: &ctx::Ctx,
-        pool: &ConnectionPool,
-    ) -> anyhow::Result<()> {
+    /// Creates a new `BlockStore` for the underlying `ConnectionPool`.
+    pub fn store(&self) -> BlockStore {
+        Store::new(self.pool.clone(), self.operator_address).into_block_store()
+    }
+
+    // Wait for all pushed miniblocks to be produced.
+    pub async fn sync(&self, ctx: &ctx::Ctx) -> ctx::Result<()> {
```

Comment:
Bikeshedding: I'd rename this method to something like `wait_for_miniblocks()` for clarity.